services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -180,50 +171,39 @@ public void setUp() {
      * @throws IOException if there is any error in reading the file.
      */
     @Test
-    public void testSaveBulkSubmission() throws IOException {
-        SdtContext.getContext().setRawInXml(Utilities.getRawXml("src/unit-test/resources/", "testXMLValid2.xml"));
+    void testSaveBulkSubmission() throws IOException {
+        SdtContext.getContext().setRawInXml(Utilities.getRawXml(PATH_TEST_RESOURCES, "testXMLValid2.xml"));
 
         // Activate Mock Generic Dao
         final IBulkSubmission bulkSubmission = this.createBulkSubmission();
         mockGenericDao.persist(bulkSubmission);
-        EasyMock.expectLastCall();
 
         // Mock the serviceRequest fetch
         final IServiceRequest serviceRequest = new ServiceRequest();
-        EasyMock.expect(mockGenericDao.fetch(IServiceRequest.class, 1)).andReturn(serviceRequest);
-
-        // Replay the EasyMock
-        EasyMock.replay(mockGenericDao);
-
-        EasyMock.expect(mockBulkCustomerDao.getBulkCustomerBySdtId(10)).andReturn(bulkSubmission.getBulkCustomer());
+        when(mockGenericDao.fetch(IServiceRequest.class, 1)).thenReturn(serviceRequest);
 
-        EasyMock.replay(mockBulkCustomerDao);
+        when(mockBulkCustomerDao.getBulkCustomerBySdtId(10)).thenReturn(bulkSubmission.getBulkCustomer());
 
         final String key =
                 bulkSubmission.getBulkCustomer().getSdtCustomerId() + bulkSubmission.getCustomerReference();
 
         // Setup concurrency map as if validator had done it.
         IInFlightMessage inFlightMessage = new InFlightMessage();
-        inFlightMessage.setCompetingThreads(new HashMap<Thread, Thread>());
+        inFlightMessage.setCompetingThreads(new HashMap<>());
         inFlightMessage.getCompetingThreads().put(Thread.currentThread(), Thread.currentThread());
 
-        EasyMock.expect(mockConcurrencyMap.get(key)).andReturn(inFlightMessage);
-        EasyMock.expect(mockConcurrencyMap.put(key, inFlightMessage)).andReturn(inFlightMessage);
-        EasyMock.replay(mockConcurrencyMap);
+        when(mockConcurrencyMap.get(key)).thenReturn(inFlightMessage);
 
         // Put a dummy value into the SdtContext
         SdtContext.getContext().setServiceRequestId(1L);
 
         // Call the bulk submission service
         bulkSubmissionService.saveBulkSubmission(bulkSubmission);
 
-        LOGGER.debug("bulkSubmission[" + bulkSubmission + "]");
-
         // Verify the Mock
-        EasyMock.verify(mockGenericDao);
-
-        Assert.assertTrue("Expected to pass", true);
+        verify(mockGenericDao, times(2)).persist(any());
```

Comment:
I'm probably not following the code correctly but why is this checking that persist() is invoked twice?  This method only sets up mockGenericDao to invoke it once.  Should it also check for the exact object that was persisted rather than using any()?

Reply:
we don't need to call times(2), changed this and replaced with exact object

Reply:
Confirmed removal of persist and use of exact object.